marketplaceItems/questionApp/appResources/appData/question_app.js

```diff
@@ -0,0 +1,313 @@
+    function createQuestionMark() {
+        if (questionMark) {
+            return;
+        }
+        parentJointIndex = MyAvatar.getJointIndex("Hips");
+        if (parentJointIndex === -1) {
+            parentJointIndex = 0;
+        }
+        var entitySpawnPosition = MyAvatar.position;
+        entitySpawnPosition.y = entitySpawnPosition.y + HALF * MyAvatar.getHeight() + HALF * 
+            QUESTION_MARK_START_DIMENSIONS_M.y + Y_OFFSET_HEAD_TOP_TO_ENTITY_M;
+        var questionMarkProperties = {
+            name: QUESTION_MARK_PROPERTY_NAME,
+            type: "Model",
+            modelURL: Script.resolvePath("resources/models/questionMark.fbx"),
+            lifetime: 360000,
+            parentID: MyAvatar.sessionUUID,
+            parentJointIndex: parentJointIndex,
+            localRotation: Quat.fromVec3Degrees({ x: 0, y: 180, z: 0 }),
+            position: entitySpawnPosition,
+            dimensions: QUESTION_MARK_START_DIMENSIONS_M,
+            grab: { grabbable: false },
+            collisionless: true
+        };
+        questionMark = Entities.addEntity(questionMarkProperties, 'avatar');
+        changingInterval = Script.setInterval(function() {
+            if (changingInterval) {
+                var questionMarkDimensions = Entities.getEntityProperties(questionMark, 'dimensions').dimensions;
+                if (questionMarkDimensions.y < MAX_HEIGHT_M) {
+                    questionMarkDimensions = Vec3.multiply(questionMarkDimensions, GROWTH_RATIO);
+                    var entitySpawnPosition = MyAvatar.position;
+                    entitySpawnPosition.y = entitySpawnPosition.y + HALF * MyAvatar.getHeight() + HALF * 
+                        questionMarkDimensions.y + Y_OFFSET_HEAD_TOP_TO_ENTITY_M;
+                    Entities.editEntity(questionMark, { 
+                        dimensions: questionMarkDimensions,
+                        position: entitySpawnPosition
+                    });
+                    lastDimensions = questionMarkDimensions;
+                } else {
+                    Script.clearInterval(changingInterval);
+                    changingInterval = null;
+                }
+            }
+        }, GROWTH_INTERVAL_MS);
+    }
+
+    /* ON CLICKING APP BUTTON: (on the toolbar or tablet) if we are opening the app, play a sound and get the question mark.
+    If we are closing the app, remove the question mark and play a different sound */
+    var OPEN_SOUND = SoundCache.getSound(Script.resolvePath('resources/sounds/open.mp3'));
+    var OPEN_SOUND_VOLUME = 0.2;
+    var CLOSE_SOUND = SoundCache.getSound(Script.resolvePath('resources/sounds/close.mp3'));
+    var CLOSE_SOUND_VOLUME = 0.3;
+    var QUESTION_CHANNEL = "QuestionChannel";
+    var messagesReceivedConnected;
+    var scaleChangedConnected;
+    var skeletonChangedConnected;
+    function onClicked() {
+        if (questionMark) {
+            cleanUp();
+            if (messagesReceivedConnected) {
```

Comment:
The code in lines 117 through 128 show up three times in this file. Could you please move that code to a function (perhaps called `disconnectSignalHandlers()` ?) and DRY up this code?